### PR TITLE
Enable tests that depend on the presence of unmanaged callers only in the runtime

### PR DESCRIPTION
**Pull request by 333fred**
Now that we have .NET 5 in our test assemblies, we can enable tests that actually need it to run. Fixes https://github.com/dotnet/roslyn/issues/46676.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs [hunk 1]
```diff
@@ -6140,44 +6135,84 @@ public void Override_ConventionOrderingDoesNotMatter()
...
+Derived1 Stdcall, Thiscall return
+Derived1 Stdcall, Thiscall ref param
+Derived1 Stdcall, Thiscall ref return
+Derived2 Stdcall, Stdcall, Thiscall param
+Derived2 Stdcall, Stdcall, Thiscall return
+Derived2 Stdcall, Stdcall, Thiscall ref param
+Derived2 Stdcall, Stdcall, Thiscall ref return
+";
+
+            var allSourceComp = CreateCompilationWithFunctionPointers(@using + executableCode + source1 + source2, options: TestOptions.UnsafeReleaseExe);
```
Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs:6207]: Consider passing as a collection of source: `CreateCompilationWithFunctionPointers(new[] { executableCode, source1, ... }, ...)`.

**Review by cston — APPROVED** [review]

**Comment by AlekseyTs** [comment]
It looks like there are CI failures.

**Comment by jaredpar** [comment]
Looks like the test failures are #50521.

**Review by jaredpar — APPROVED** [review]

**Comment by 333fred** [comment]
> Looks like the test failures are #50521. 

Aleksey was commenting on the test failures I had fixed locally but forgotten to push.

**Comment by 333fred** [comment]
@AlekseyTs I've addressed the test failures.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 4)

**Pull request merged**